tests/Unit/Css/CssDocumentTest.php

```diff
@@ -516,13 +583,18 @@ public function provideInvalidNonConditionalAtRule(): array
      */
     public function notRendersAtMediaRuleInNonConditionalAtRules(): void
     {
-        $subject = new CssDocument('@media screen { p { color: red; } }');
+        $subject = self::createDebugSubject('@media screen { p { color: red; } }');
 
         $result = $subject->renderNonConditionalAtRules();
 
         self::assertSame('', $result);
     }
 
+    private static function createDebugSubject(string $css): CssDocument
```

Comment:
Please let's have this method at the top directly below `setUp`/`tearDown` as this  method is closely related to `setUp`.

Reply:
Agree.  Wasn't sure where to put it, but have given this a bit of thought - see my other comment.  Moved.

Now even wondering if constants and properties need necessarily be placed at the top of the class, or could be placed close to where they are relevant...

Reply:
AFAIK, the recommended order (which also is enforced by some style checker rules IIRC) for "things" in a class is:

1. using traits
2. class constants
3. properties
4. methods

Reply:
I remember seeing this before.  Perhaps you posted it here.  It is logical.  I was pushing the boundaries by suggesting methods could be interleaved with properties (but why not?).  But on the other side of the coin are the sticklers for insisting that all `static` methods be grouped together, then they be grouped by `public`/`protected`/`private`.  I have seen the unmaintainability of that, e.g. when the signature of a method needs changing.  But what about when a property needs to be changed into a method (or, rarely, vice versa)?

These are slightly rhetorical thoughts and mostly meant as food for thought for your Christmas dinner.  Have a good one 😊 ❤️@🎄.